lib/src/utf16.dart

```diff
@@ -22,12 +22,9 @@ extension Utf16Pointer on Pointer<Utf16> {
   /// The UTF-16 code units of the strings are the non-zero code units up to
   /// the first zero code unit.
   int get length {
-    final Pointer<Uint16> array = cast<Uint16>();
-    int length = 0;
-    while (array[length] != 0) {
-      length++;
-    }
-    return length;
+    _ensureNotNullptr('length');
+    final codeUnits = cast<Uint16>();
+    return _length(codeUnits);
```

Comment:
Splitting it out into a separate function isn't technically needed for UTF-16 (because toDartString doesn't compute `length` like it does for UTF-8).
Still OK abstraction.